### ISSUE TITLE
Adding "Stratify the train set by"

### ISSUE THREAD

**Issue by jbrion38** [emilienschultz/activetigger#302]
Following yesterday's discussion, you told me about the “Stratify the test set by” feature and I thought it would be interesting to be able to do the same thing on the train set (in my case, I have more articles from Ouest France than from Le Monde, so it might be interesting to stratify according to my newspaper category).

**Comment by emilienschultz** [comment]
before starting to implement it, small question : Does the situation exist we want a different stratification on the testset and on the trainset (@jboelaert)

If no, we have 3 different situations when we create a project, no ?  
- deterministic selection of the N first elements without creating a testset
- no stratification, and random selection of elements both in train/test
- stratification with random selection under this constraint both in train/test

**Comment by jboelaert** [comment]
I don't think one would ever want to stratify train and test differently,
so let's go as you suggest.
If someone really wanted to stratify differently, they could always use
"deterministic" on their full training data, and separately upload their
test set.

**Comment by emilienschultz** [comment]
Yep, great, I will do this way then ! Thanks

**Comment by emilienschultz** [comment]
I updated the selection and for the moment removed the automatic enforced selection of annotated data before sampling, because it can create strange behaviour
